diff -Naur iputils-s20100214//ping6.c iputils-s20100214.tpg//ping6.c
--- iputils-s20100214//ping6.c	2010-02-13 18:39:23.000000000 +0000
+++ iputils-s20100214.tpg//ping6.c	2010-04-18 01:12:25.000000000 +0000
@@ -403,10 +403,10 @@
 			unsigned int iface = if_name2index(device);
 #endif
 #ifdef IPV6_RECVPKTINFO
-			struct in6_pktinfo ipi;
+			struct in6_pktinfo *ipi;
 
 			memset(&ipi, 0, sizeof(ipi));
-			ipi.ipi6_ifindex = iface;
+			ipi->ipi6_ifindex = iface;
 #endif
 
 #ifdef HAVE_SIN6_SCOPEID
